### PR TITLE
Started setting the FlutterOverlayView's bounds when it gets added to its superview.

### PR DESCRIPTION
**Pull request by gaaclarke**
Related issue:https://github.com/flutter/flutter/issues/45443

This crash was happening because the FlutterOverlayView is getting removed and added to its superview so it is missing its opportunity to do layout.  See the related issue for more details.

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm [hunk 1]
```diff
@@ -370,10 +370,13 @@
   for (int64_t view_id : composition_order_) {
     EnsureOverlayInitialized(view_id, gl_context, gr_context);
     auto frame = overlays_[view_id]->surface->AcquireFrame(frame_size_);
-    SkCanvas* canvas = frame->SkiaCanvas();
-    canvas->drawPicture(picture_recorders_[view_id]->finishRecordingAsPicture());
-    canvas->flush();
-    did_submit &= frame->Submit();
+    if (frame) {
+      // We should already be printing an error if we got here.
```
Comment (amirh) [inline shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm]: I'm a little confused by this comment, can you elaborate a bit more?

Comment (gaaclarke) [reply shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm]: `AcquireFrame` returning null is an error.  From my experience, if it returns null, it prints out an error message.  I don't want to print out 2 error messages for the same error.

Comment (gaaclarke) [reply shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm]: For example: https://github.com/flutter/engine/blob/a86ef946563b020108320bbfb974bf7343284fd3/shell/gpu/gpu_surface_gl.cc#L264

Comment (gaaclarke) [reply shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm]: I updated it, it was in the wrong place.

Comment (amirh) [reply shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm]: Thanks makes more sense to me now.

**Comment by amirh** [comment]
Overall looks good, let's add a test that rotates the device.

**Comment by gaaclarke** [comment]
> Overall looks good, let's add a test that rotates the device.

Thanks @amirh, do you know how we'd do something like that?  It seems like we'd want a golden image test that rotates the screen.
1) We don't have golden image tests for the engine.
1) Can golden image tests rotate the screen?  If we are running them on device lab that won't be possible without some robotics ;)

**Comment by gaaclarke** [comment]
After further thought, I don't think you can write a test for this without a robot arm since there is no programatic way to rotate screens.

**Comment by amirh** [comment]
We do have golden image test for iOS platform views, see [here](https://github.com/flutter/engine/blob/master/testing/scenario_app/ios/Scenarios/ScenariosUITests/PlatformViewUITests.m). It runs on CI on a simulator.

Seems like we may be able to rotate the simulator in these tests by setting the [orientation](https://developer.apple.com/documentation/xctest/xcuidevice/1619046-orientation?language=objc) of [XCUIDevice.sharedDevice](https://developer.apple.com/documentation/xctest/xcuidevice/1619048-shareddevice?language=objc)? (I didn't try it, but I'd assume XCUITest lets you rotate the simulator, and this is what came up when googling)

Seems like the codepath for the fix is active with software rendering as well, so the simulator test should be a reasonable coverage?

**Comment by gaaclarke** [comment]
I've added a golden image test.  It is doing what I asked it to do.  Let me double check it tomorrow to verify that it is exercising this PR.  I'll ping you tomorrow when I've verified that.

**Comment by gaaclarke** [comment]
@amirh I reverted the golden test.  I was unable to reproduce the error in the tests.  I suspect there is a timing issue related to when you rotate the screen and XCUIDevice is rotating with baby gloves on.  It is also possible that it doesn't manifest on the simulator for some reason.

**Comment by gaaclarke** [comment]
Talked to @amirh offline, the non-perfect test is preferred.

**Pull request merged**